corehq/apps/users/static/users/js/roles.js

```diff
@@ -192,9 +193,37 @@ hqDefine('users/js/roles',[
             },
         };
 
+        self.allowSubmit = ko.computed(function () {
+            if (!self.roleBeingEdited()) {
+                return false;
+            }
+            if (!self.roleBeingEdited().name()) {
+                return false;
+            }
+            return true;
+        });
+
+        self.submitNewRole = function () {
+            // moved saveOptions inline
+            $.ajax({
+                method: 'POST',
+                url: o.saveUrl,
+                data: JSON.stringify(UserRole.unwrap(self.roleBeingEdited)),
+                dataType: 'json',
+                success: function (data) {
+                    self.addOrReplaceRole(data);
+                    self.unsetRoleBeingEdited();
+                },
+                error: function (data) {
+                    console.log('error');
```

Comment:
It'd be helpful for this to show the user there was a problem. One easy thing to do is use the `alert_user` library, which will display a message at the top of the page. One example: https://github.com/dimagi/commcare-hq/blob/d2da78ecdb8660ab5a185c413ead2060362f4b86/corehq/apps/case_search/static/case_search/js/case_search.js#L62-L64